Python/ast_unparse.c

```diff
@@ -58,6 +59,18 @@ append_charp(_PyUnicodeWriter *writer, const char *charp)
         } \
     } while (0)
 
+static PyObject *
+replace_inf_value(PyObject *repr)
+{
+    PyObject *inf_str = PyUnicode_InternFromString("inf");
```

Comment:
You need to check for failures on all these calls and manage errors in these calls.